module/os/fleet.py

```diff
@@ -66,7 +66,8 @@ def _walk_sight(self):
         return sight
 
     def handle_mystery(self, button=None):
-        return False
+        # Treat map events as mystery, may walk into unexpected object on path
+        return self.handle_map_event()
```

Comment:
Map event is already handled in `handle_ambush()`, so here is redundant.
Also, if fleet encountered something in walk path, `handle_ambush()` will trigger a walk retry.
If it's handled in `handle_mystery()`, it won't trigger a walk retry, and finally cause walk timeout.

Reply:
Well I don't have the log anymore but I did indeed encounter an infinite loop of this printing statement that was added to camera.py.
```
elif self.appear(GET_ITEMS_1): logger.warning('Items got. Trying handling mystery.') self.handle_mystery() return self.update(camera=camera)
```
So it seemed like that was the best way to handle it. But if its as you say no problem. If I do encounter it again I will report and submit the error log next time.